src/fairseq2/data/parquet/configs.py

```diff
@@ -0,0 +1,351 @@
+    """
+
+    name: Optional[str] = None
+    """When name is provided, it will use preregistered cards to populate all attributes.
+        name convention is the following
+        -  {card_name}={split}:{weight}
+
+        Example:
+        - wiki
+        - wiki:0.2 # no split
+        - wiki=dev  # default weight=1
+        - wiki=dev:0.2
+
+        Cards attributes will be overwritten by user defined ParquetDatasetConfig in
+        `create_dataset_config_from_cards`.
+    """
+
+    parquet_path: str = ""
+    """The path to parquet dataset file.
+    """
+
+    weight: float = 1.0
+    """
+    Indicates relative weight of dataset that can be used for sampling from different datasets.
+    """
+
+    limit: Optional[ParquetDatasetLimitOptions] = None
+    """
+    Contains different options that allows to load only a part of the provided dataset.
+    It will **always** take some number of **first** fragments according to the order in which
+    they appear in the dataset and this logic will not be depedent on suffling/seed.
+    When several limits are provided, each of them will be applied (resulting in the strongest limit).
+    """
+
+    partition_filters: Optional[str] = None
+    """
+    Filters that should be applied only on partition columns for fast partition prunning.
+    This filters should not be duplicated in `filters` (below) which are used on materialized data.
+    To know the partition columns on dataset :
+    ```python
+    >>> pq.ParquetDataset(parquet_path).partitioning.schema.names
+    ```
+    Note that for if `parquet_path` references a single file -> the result above will NOT be correct (returns all columns).
+    Note that for a single file case, there should no partition_filters since there're no partitions !!
+    """
+
+    filters: Optional[Union[List[Any], pa.dataset.Expression]] = None
+    """
+    This can be any valid pyarrow dataset filter expression, or a list of old-style tuples
+    that we convert to an expression in __post_init__.
+
+    See https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Expression.html#pyarrow.dataset.Expression
+
+    Some examples :
+
+    >>> import pyarrow.compute as pc
+    >>> import pyarrow as pa
+
+    >>> filters = [("data_split", "=", "train"), ("lang1", "in", ["eng","spa"]), ("lang2", "=", "eng")])
+    >>> filters = (pc.field("data_split") == pc.scalar("train")) & (pc.field("duration") > 7)
+    >>> filters = pa.compute.greater(pa.compute.utf8_length(ds.field("lang1_text")), 4)
+    >>> filters = pa.compute.less_equal(pa.compute.list_value_length(pa.dataset.field("audio_wav")), 16_000 * 30)
+
+    Note that all fields used here should be among existing columns in the dataset schema.
+    """
+
+    split_to_row_groups: bool = True
+    """If ``True``, uses Parquet row groups instead of simple partitions which
+    are generally smaller. Highly recommended for non-partitioned parquet files."""
+
+    nb_parallel_fragments: Optional[int] = 5
```

Comment:
maybe we could keep it with default=None and add an extra config arg (max_tokens) to be used with dynamic bucketing.